From e9b1243f14689a312489da2de5d47b4554425f0b Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@gmail.com>
Date: Fri, 29 Sep 2023 20:40:25 +0200
Subject: [PATCH] google: add gpr lib to TTS_ENGINE call

* with new grpc from:
  dfac3cc512 devtools: grpc: bump to 1.56.2

  this was now failing with:
/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: CMakeFiles/tts-service.dir/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/include/google/cloud/texttospeech/v1/cloud_tts.grpc.pb.cc.o: undefined reference to symbol 'gpr_inf_future'
/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: /OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/lib/libgpr.so.33: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.

  unfortunately this hardcodes list of libs instead of using .cmake
  which lists it correctly:
./recipe-sysroot/usr/lib/cmake/grpc/gRPCTargets-noconfig.cmake:  IMPORTED_LOCATION_NOCONFIG "${_IMPORT_PREFIX}/lib/libgpr.so.33.0.0"
./recipe-sysroot/usr/lib/cmake/grpc/gRPCTargets-noconfig.cmake:  IMPORTED_SONAME_NOCONFIG "libgpr.so.33"
./recipe-sysroot/usr/lib/cmake/grpc/gRPCTargets-noconfig.cmake:list(APPEND _cmake_import_check_files_for_gRPC::gpr "${_IMPORT_PREFIX}/lib/libgpr.so.33.0.0" )

set_target_properties(gRPC::grpc PROPERTIES
  INTERFACE_COMPILE_FEATURES "cxx_std_14"
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include"
  INTERFACE_LINK_LIBRARIES "ZLIB::ZLIB;c-ares::cares;gRPC::address_sorting;re2::re2;gRPC::upb;dl;m;Threads::Threads;rt;absl::cleanup;absl::flat_hash_map;absl::flat_hash_set;absl::inlined_vector;absl::bind_front;absl::function_ref;absl::hash;absl::type_traits;absl::statusor;absl::span;absl::utility;gRPC::gpr;OpenSSL::SSL;OpenSSL::Crypto;gRPC::address_sorting;gRPC::upb"
)

* similarly absl_synchronization:

/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: CMakeFiles/tts-service.dir/src/engines/tts/google/GoogleTTSEngine.cpp.o: undefined reference to symbol '_ZN4absl12lts_202308025MutexD1Ev'
/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: /OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/lib/libabsl_synchronization.so.2308.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status

/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: CMakeFiles/tts-service.dir/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/include/google/api/metric.pb.cc.o: undefined reference to symbol '_ZN4absl12lts_2023080213hash_internal15MixingHashState5kSeedE'
/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: /OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/lib/libabsl_hash.so.2308.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status

/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: CMakeFiles/tts-service.dir/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/include/google/cloud/texttospeech/v1/cloud_tts.grpc.pb.cc.o: undefined reference to symbol '_ZN4absl12lts_202308024Cord6AppendEOS1_'
/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: /OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/lib/libabsl_cord.so.2308.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status

/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: CMakeFiles/tts-service.dir/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/include/google/cloud/texttospeech/v1/cloud_tts.grpc.pb.cc.o: undefined reference to symbol '_ZN4absl12lts_2023080213cord_internal17cordz_next_sampleE'
/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: /OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/lib/libabsl_cordz_functions.so.2308.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status

/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: CMakeFiles/tts-service.dir/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/include/google/api/metric.pb.cc.o: undefined reference to symbol '_ZN4absl12lts_2023080212log_internal15LogMessageFatalC1EPKci'
/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: /OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/lib/libabsl_log_internal_message.so.2308.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status

/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: CMakeFiles/tts-service.dir/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/include/google/api/metric.pb.cc.o: undefined reference to symbol '_ZN4absl12lts_2023080212log_internal9kCharNullE'
/OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot-native/usr/bin/x86_64-webos-linux/../../libexec/x86_64-webos-linux/gcc/x86_64-webos-linux/13.2.0/ld: /OE/build/luneos-nanbield/tmp-glibc/work/corei7-64-webos-linux/com.webos.service.tts/1.0.0-28+git/recipe-sysroot/usr/lib/libabsl_log_internal_nullguard.so.2308.0.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status

Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
---
Upstream-Status: Pending

 src/engines/tts/google/CMakeLists.txt | 9 ++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/engines/tts/google/CMakeLists.txt b/src/engines/tts/google/CMakeLists.txt
index f04a47d..9b17f2d 100755
--- a/src/engines/tts/google/CMakeLists.txt
+++ b/src/engines/tts/google/CMakeLists.txt
@@ -26,6 +26,15 @@ set(src ${CMAKE_CURRENT_SOURCE_DIR}/GoogleTTSEngine.cpp
     ${GOOGLE_SOURCE}
 )
  set(deps
+     gpr
+     absl_cord
+     absl_cordz_info
+     absl_cordz_functions
+     absl_log_internal_message
+     absl_log_internal_nullguard
+     absl_synchronization
+     absl_hash
+     absl_city
      grpc
      grpc++
      protobuf
